### ISSUE TITLE
SpectatorView Deployment on iOS Device

### ISSUE THREAD

**Issue by ryanxdebelen** [Microsoft/MixedRealityToolkit#183]
I am trying to build the SpectatorViewExample scene for an iOS device and I get several errors below:

Assets/HoloToolkit-Preview/QRTracker/Scripts/QRCodesManager.cs(84,23): error CS0012: The type `System.MulticastDelegate' is defined in an assembly that is not referenced. Consider adding a reference to assembly `System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

Assets/HoloToolkit-Preview/QRTracker/Scripts/SpatialGraphCoordinateSystem.cs(42,45): error CS0117: `UnityEngine.XR.WSA.WorldManager' does not contain a definition for `OnPositionalLocatorStateChanged'

Assets/HoloToolkit-Preview/QRTracker/Scripts/SpatialGraphCoordinateSystem.cs(43,60): error CS0117: `UnityEngine.XR.WSA.WorldManager' does not contain a definition for `state'

I am using Experimental (.NET 4.6 Equivalent) for scripting runtime version. If I switch to Stable (.NET 3.5 Equivalent), the same error is being thrown with the System.Runtime, Version=2.0.0.0 instead.

Thank you very much for your help.

**Comment by CommanderLee** [comment]
Based on my experiment, QRTracker is only used in the HoloLens side, so you might move this folder away when you build the iOS version, as you did for the GLTF folder. 

I am not sure whether it will affect other functions by doing this, but at least I can run the sample scene now.

**Comment by ryanxdebelen** [comment]
Thank you for your reply.

I figured that out and decided to use a version of MRTK which does not contain the QRTracker folder. That fixed the issue, but I ran into an issue with my SpectatorViewerPlugin, specifically with all the .dlls generated by the SpectatorViewPlugin sln:


**Sample:**
 Loading opencv_features2d341.dll

Failed to load 'opencv_features2d341.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

I did follow the instructions in [](https://github.com/Microsoft/MixedRealityToolkit/blob/master/SpectatorViewPlugin/README.md) and built the solution with an x86 configuration. 

Could you please tell me the versions of the SDKs you are using?

Is there something that I am missing here?

Thank you for your help.

**Comment by ryanxdebelen** [comment]
Thank you for your reply.

I figured that out and decided to use a version of MRTK which does not contain the QRTracker folder. That fixed the issue, but I ran into an issue with my SpectatorViewerPlugin, specifically with all the .dlls generated by the SpectatorViewPlugin sln:

Sample:
Loading opencv_features2d341.dll

Failed to load 'opencv_features2d341.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

I did follow the instructions in [https://github.com/Microsoft/MixedRealityToolkit/blob/master/SpectatorViewPlugin/README.md](url)
 and built the solution with an x86 configuration.

Could you please tell me the versions of the SDKs you are using?

Is there something that I am missing here?

Thank you for your help.

**Comment by CommanderLee** [comment]
Hi @ryanxdebelen , did you read very first issue in this thread (https://github.com/Microsoft/MixedRealityToolkit/issues/188)? That post helped me a lot.

Basically, we got a few OpenCV-UWP dlls by running the bash commands following the instruction, but all these dlls we got were OpenCV "343" versions. The VS solution made a mistake and linked "341" versions. 

So you need to check the "linker" settings in the VS project settings and change these numbers manually.

**Comment by CommanderLee** [comment]
I misunderstood your post just now so I deleted my previous reply.

For your problem, I don't know. I am using Windows 10 17763 SDK + Unity 2017 LTS + VS 2017. I haven't tried other versions. BTW, I followed the instructions, and I got "343" version dlls. Then I created the directory as suggested, and move these dlls to the new directory. Finally I opened the Unity example scene, and that's it.

**Comment by ryanxdebelen** [comment]
Oh, that's weird. I got the "341" version dlls when I followed the instructions.

However, when I manually changed the additional dependencies to include the "343" version dlls, an error was thrown: LINK : fatal error LNK1181: cannot open input file 'opencv_aruco343.lib'

It seems that there are new changes. I am trying to run it again and see if it it will work this time.

With regards to the SDKS, are you using the latest version from this?
[https://github.com/Microsoft/MixedRealityToolkit-Unity](url)

Thank you for your help.

**Comment by ryanxdebelen** [comment]
Update:

I followed the instructions again, got the "343" version dlls, and successfully built the SpectatorViewPlugin. Afterwards, I created the directory as suggested, and moved the generated dlls to the new directory. The previous issue (relating to the .dlls):

**Sample:**
Loading opencv_features2d341.dll

Failed to load 'opencv_features2d341.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

got **RESOLVED**. However, when I deploy it to HoloLens, it says: 

  Loading SpectatorViewPlugin.dll

Failed to load 'SpectatorViewPlugin.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

I do believe that this has an issue with the configuration manager when I compile my plugin.

Although it says that Release|x86 for active solution configuration and active solution platform, when I open the configuration manager, the platform says that it is Win32. There are no other options available. See below:

![image](https://user-images.githubusercontent.com/29617854/48234590-47330080-e40e-11e8-96e0-eec66c6d133f.png)

Even though I try to make a new solution platform, x86 is not in the options. I wonder if there is any difference between x86 and win32 platform in this case.

**Comment by ryanxdebelen** [comment]
Update:

I followed the instructions again, got the "343" version dlls, and successfully built the SpectatorViewPlugin. Afterwards, I created the directory as suggested, and moved the generated dlls to the new directory. The previous issue (relating to the .dlls):

Sample:
Loading opencv_features2d341.dll

Failed to load 'opencv_features2d341.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

got RESOLVED. However, when I deploy it to HoloLens, it says:

Loading SpectatorViewPlugin.dll

Failed to load 'SpectatorViewPlugin.dll', expected x86 architecture, but was Unknown architecture. You must recompile your plugin for x86 architecture.

I do believe that this has an issue with the configuration manager when I compile my plugin.

Although it says that Release|x86 for active solution configuration and active solution platform, when I open the configuration manager, the platform says that it is Win32. There are no other options available. See below:

![image](https://user-images.githubusercontent.com/29617854/48234658-92e5aa00-e40e-11e8-822d-531be6071eff.png)

Even though I try to make a new solution platform, x86 is not in the options. I wonder if there is any difference between x86 and win32 platform in this case.
